onnxruntime/core/optimizer/nhwc_transformer.cc

```diff
@@ -74,7 +74,8 @@ Status NhwcTransformer::ApplyImpl(Graph& graph, bool& modified, int graph_level,
   }
 
   if (modified) {
-    Optimize(*api_graph, /*allow_extended_ops*/ true, kCpuExecutionProvider);
+    Optimize(*api_graph, /*allow_extended_ops*/ true, kCpuExecutionProvider, OptimizerMode::OPTIMIZE_TRANSPOSE,
+             OrtDefaultCostCheck);
```

Comment:
there's another use of Optimize() here:
https://github.com/microsoft/onnxruntime/blob/24ad202ec6b4926b7ab25838d2ddfc20cd5ea1fc/onnxruntime/core/optimizer/transpose_optimizer/ort_transpose_optimizer.cc#L20

should that one use OrtDefaultCostCheck or an empty CostCheckFn? if OrtDefaultCostCheck is a reasonable default (I'm just assuming this from the name), should that be the default Optimize() cost_check_fn value instead?

Reply:
The core transpose optimizer including Optimize is meant to be generic, so we don't want the default cost check to be an ORT specific one.

As this is an L1 transformer it won't make a difference though as no nodes are assigned yet, and the ORT checker special cases require nodes to be assigned to an ORT EP to kick in.

Reply:
I see, thanks for the clarification. is OrtDefaultCostCheck intended to only be applicable for L2+ transformers? perhaps a comment or more specific name for it would help?